### ISSUE TITLE
Combine no longer works

### ISSUE THREAD

**Issue by EliSnow** [pipelinedb/pipelinedb#1924]
I am in the process of migrating a lot of data from 0.9.7 to 0.9.8 and have discovered that `combine` no longer works.

```sql
CREATE STREAM s (n int);

CREATE CONTINUOUS VIEW cv AS
  SELECT avg(n)
  FROM s;

SELECT combine(avg) FROM cv;
```
results in the following error:
```
ERROR:  the FROM clause is not a continuous view
HINT:  Only aggregate continuous view columns can be combined.
```

`PipelineDB 0.9.8 at revision d1c40b1bda8de58ae27cca84acc15478e8c4147f on x86_64-pc-linux-gnu, compiled by gcc (Debian 6.3.0-18) 6.3.0 20170516, 64-bit`

Am I missing something obvious? I didn't see any notes about this in the 0.9.8 release blog, but it seems like something this big would have been caught by CI tests.

**Comment by EliSnow** [comment]
Ok, I did a bunch more testing and narrowed it down to an incompatability between pg_stat_statements and PipelineDB 0.9.8. I did test that when I change to v0.9.7 their is no incompatability and everything works as expected.

The relevant GUCs I use for pg_stat_statements are:

```
shared_preload_libraries = 'pg_stat_statements'
pg_stat_statements.track = all
```

I am guessing in the work to refactor to an extension something was changed such that when pg_stat_statement executes it's hooks during query execution PipelineDB no longer recognizes the `CONTINUOUS VIEW` in the `FROM` clause.

**Comment by derekjn** [comment]
@EliSnow gotcha nice work! We'll take a look at this, I'm guessing we should be able to provide better compatibility here.

**Comment by derekjn** [comment]
@EliSnow this appears to not be an issue in `master`. Is upgrading to `0.9.9` an option for you? We're ready to release it now and actually wanted to see if we could get any necessary fixes for this issue into `0.9.9` but as I mentioned it's not an issue in the latest release...

**Comment by EliSnow** [comment]
Only if there's no catalog changes... Also since the database in question is currently undergoing a data-migration (through a `COPY` from the old database) I don't want to touch it till it's finished. My estimate for it finishing will be around next Monday.

Though I could always pull down and test 0.9.9 on my local machine and test that it's working.

**Comment by derekjn** [comment]
@EliSnow there are catalog changes but you can just use the [binary upgrade](http://docs.pipelinedb.com/upgrades.html#upgrades) tool pretty easily so that shouldn't be a big concern.

**Comment by EliSnow** [comment]
Perhaps I misunderstand the documentation but this line

> and then simply copies all other database data at the operating system level

leaves me to believe that all the data needs to be copied on disk. If so it is still prohibitive in my case, because we'd need to double the size of our EBS volume and once you grow it you can't shrink it back down due to AWS limitations, then we're stuck paying twice as much for many TBs of unused SSD storage.

Or is `pipeline-upgrade` creating hard links?

**Comment by derekjn** [comment]
> leaves me to believe that all the data needs to be copied on disk. If so it is still prohibitive in my case, because we'd need to double the size of our EBS volume and once you grow it you can't shrink it back down due to AWS limitations, then we're stuck paying twice as much for many TBs of unused SSD storage.

You're correct about additional disk space requirements for binary upgrades. It's fair to think of a binary upgrade as a data directory copy at the OS level.

That being said, there are no storage-level changes with the `0.9.9` release, so you should still be able to dump/restore CV data from older releases into a `0.9.9` DB.

**Comment by agentgt** [comment]
@EliSnow I'm curious if and when you upgrading if you see #1902 happen.

**Comment by EliSnow** [comment]
I wouldn't because I'm not using `pipeline-upgrade`.

**Comment by EliSnow** [comment]
@derekjn, I just tried using `pipeline-upgrade` to migrate from 0.9.8 to 0.9.9 (because apparently merely removing the pg_stat_statements extension after you experience this bug does not resolve it) but `pipeline-upgrade` cannot work when you have a CV with a column utilizing a composite `first_values` (ie `first_values(10) WITHIN GROUP (ORDER BY col1, col2, col3)`).

I get a failure error from `pipeline_upgrade` and the error in the logs state:
```
could not execute query: ERROR:  column "values" has pseudo-type record[]
```

Would you like me to create a separate bug for that? Unless there's a workaround I guess I'm stuck migrating data over the slow way.

**Comment by derekjn** [comment]
@EliSnow we're going to repro this issue on our end to see if we can identify a workaround for you or potentially publish a release update. I'll follow up here when we know more.

**Comment by derekjn** [comment]
@EliSnow we've been looking at this and we've identified some issues with the binary upgrade path for some environments. We should have a release update ready soon that should resolve this.

**Comment by EliSnow** [comment]
I believe the only non-required flag I used was `--link`.

**Comment by derekjn** [comment]
@EliSnow we're currently publishing release `0.9.9u0`, which we've verified works for upgrades involving `first_values` aggregates of the form you shared above. I'll let you know here when the binaries make it through the release pipeline.

**Comment by derekjn** [comment]
@EliSnow the release update has been published:

https://github.com/pipelinedb/pipelinedb/releases/tag/0.9.9u0

Please let us know if that works for you. And if you have any issues feel free to reach out to me directly on Gittr:

https://gitter.im/derekjn

I'll be around for a while.

**Comment by EliSnow** [comment]
I'll repost here what I sent to @derekjn on gitter:

Your fixed allowed me to get past that initial error I reported with pipeline-upgrade but unfortunately I ran into another.
I've got some streams which have a column with a custom, user-defined type. So I get the error `could not execute query: ERROR: type "public.nested_group[]" does not exist` because it tries to recreate the stream before recreating the custom type

**Comment by derekjn** [comment]
@EliSnow thanks I'll reach out on Gittr right now.
